csrc/kernel_cache.cpp

```diff
@@ -1610,15 +1610,14 @@ std::optional<FusionKernelRuntime::HeuristicsPtr> FusionKernelRuntime::
   FusionKernelRuntime::HeuristicsPtr heuristics =
       std::make_unique<FusionHeuristics>(num_groups);
 
-  // Store metadata copy of arguments for ArgumentManager
-  KernelArgumentHolder args_metadata = copyMetadataArg(args);
+  KernelArgumentHolder args_copy(args);
 
   // ArgumentManager manipulates the KernelArgumentHolder argument.
-  // We make another metadata copy for the PrecomputedValues.
-  KernelArgumentHolder complete_fusion_metadata_args(args_metadata);
+  // We make a copy for the PrecomputedValues.
+  KernelArgumentHolder complete_fusion_args(args_copy);
```

Comment:
Can someone teach me why this copy is required? The above comment seems to describe the copy at Line 1613 not the copy here.

Reply:
Sorry I'm still missing the context here. 

Am I right that, with this PR, `args`, `args_copy` and `complete_fusion_args` all contain actual tensors not metadata? I understood that `args_copy` is to allow `args_manager` to modify the passed-in arguments. But how does the code break if Line 1644 use `args` instead of `complete_fusion_args`?

Reply:
Yeah I think that works fine. `args_copy` is needed since `args_manager` needs a mutable reference but `PrecomputedValues::bindValues` can take the const reference `args`. I just pushed a change to remove `complete_fusion_args` and rename `args_copy` to `mut_args`.